### PR TITLE
Move FSDP README from Ronghang pr's comment (#3678)

### PR DESCRIPTION
**Pull request by JackCaoG**
Backport fsdp readme to 1.12

### CODE REVIEW

File: torch_xla/distributed/fsdp/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  --lr 0.4 --batch_size 128 --num_warmup_epochs 5 --lr_scheduler_divide_every_n_epochs 30 --lr_scheduler_divisor 10 --num_epochs 100 \
+  --use_nested_fsdp
+```
+You can also add ` --use_gradient_checkpointing` (which needs to be used along with `--use_nested_fsdp`) to apply gradient checkpointing on the residual blocks.
+
+---
+
+### Example training scripts on TPU pod (with 10 billion parameters)
+
+To train large models that cannot fit into a single TPU, one should use nested FSDP (wrapping sub-modules with inner FSDP when building the entire model) to implement the ZeRO-3 algorithm.
```
Comment (miladm) [inline torch_xla/distributed/fsdp/README.md:97]: Is there a reason we don't have perf numbers for ViT missing from this statement? I see we share it fro ResNet.

**Review by miladm — APPROVED** [review]
stamped!
Thanks.

File: torch_xla/distributed/fsdp/README.md (hunk 1 above)
Comment (JackCaoG) [reply torch_xla/distributed/fsdp/README.md:97]: it was not on ronghang's pr comment. I am just copy that over. Not sure if he has perf number.

Comment (yeounoh) [reply torch_xla/distributed/fsdp/README.md:97]: I guess that it's OK since it focuses on the usage and description of the feature/API?

**Review by yeounoh — COMMENTED** [review]
LGTM

File: torch_xla/distributed/fsdp/README.md (hunk 1 above)
Comment (miladm) [reply torch_xla/distributed/fsdp/README.md:97]: yeah, this is mostly a nit and good to have. we can go ahead without it.
